packages/pigeon/lib/pigeon_lib.dart

```diff
@@ -355,85 +355,115 @@ IOSink? _openSink(String? output) {
   return sink;
 }
 
-/// A generator that will write code to a sink based on the contents of [PigeonOptions].
-abstract class Generator {
-  /// Returns an [IOSink] instance to be written to if the [Generator] should
-  /// generate.  If it returns `null`, the [Generator] will be skipped.
-  IOSink? shouldGenerate(PigeonOptions options);
+/// An adapter that will call a generator to write code to a sink
+/// based on the contents of [PigeonOptions].
+abstract class GeneratorAdapter {
+  /// Constructor for [GeneratorAdapter]
+  GeneratorAdapter(this.fileTypeList);
+
+  /// A list of [FileType]
+  /// Used to tell the generator method which type of file to create.
+  List<FileType> fileTypeList;
+
+  /// Returns an [IOSink] instance to be written to if the [GeneratorAdapter] should
+  /// generate.  If it returns `null`, the [GeneratorAdapter] will be skipped.
```

Comment:
Nit for all of these comments: we should follow [Dart style](https://dart.dev/guides/language/effective-dart/documentation#do-separate-the-first-sentence-of-a-doc-comment-into-its-own-paragraph) about the first sentence.

Reply:
I didn't write any of these, but I can update them.